Services/source/housekeeping/housekeeping_service.c

```diff
@@ -368,26 +378,53 @@ Result mock_everyone(All_systems_housekeeping *all_hk_data) {
  * @return Result
  *      FAILURE or SUCCESS
  */
-Result collect_hk_from_devices(All_systems_housekeeping *all_hk_data) {
-    /*populate struct by calling appropriate functions*/
-    ADCS_returnState ADCS_return_code = HAL_ADCS_getHK(&all_hk_data->adcs_hk); // ADCS get housekeeeing
-    int Athena_return_code = Athena_getHK(&all_hk_data->Athena_hk);            // Athena get temperature
 
-    // EPS get housekeeping
-    EPS_getHK(&all_hk_data->EPS_hk, &all_hk_data->EPS_startup_hk);
-    UHF_return UHF_return_code = UHF_getHK(&all_hk_data->UHF_hk);      // UHF get housekeeping
-    STX_return STX_return_code = HAL_S_getHK(&all_hk_data->S_band_hk); // S_band get housekeeping
+Result collect_hk_from_devices(All_systems_housekeeping* all_hk_data) {
+  /*populate struct by calling appropriate functions*/
+  #ifndef ADCS_IS_STUBBED
+    ADCS_returnState ADCS_return_code = HAL_ADCS_getHK(&all_hk_data->adcs_hk);             //ADCS get housekeeeing
+  #endif /* ADCS Housekeeping */
+
+  #ifndef ATHENA_IS_STUBBED
+    int Athena_return_code = Athena_getHK(&all_hk_data->Athena_hk);    //Athena get temperature
+  #endif /* ADCS Housekeeping */
+  
+  #ifndef EPS_IS_STUBBED
+    eps_refresh_instantaneous_telemetry();
+    eps_instantaneous_telemetry_t eps = get_eps_instantaneous_telemetry();
+
+    //Adding these in order to get the startup telemetry - should be added elsewhere
+    eps_refresh_startup_telemetry();
+    eps_startup_telemetry_t eps_startup = get_eps_startup_telemetry();
+    EPS_getHK(&all_hk_data->EPS_hk, &all_hk_data->EPS_startup_hk);                                   //EPS get housekeeping
+  #endif /* EPS Housekeeping */
+
+  #ifndef UHF_IS_STUBBED
+    if(!uhf_is_busy()){
+        UHF_return UHF_return_code = UHF_getHK(&all_hk_data->UHF_hk);      //UHF get housekeeping
+    }
+    vTaskDelay(ONE_SECOND);
+  #endif /* UHF Housekeeping */
+
+  #ifndef SBAND_IS_STUBBED
+    STX_return STX_return_code = HAL_S_getHK(&all_hk_data->S_band_hk); //S_band get housekeeping
+  #endif /* SBAND Housekeeping */
 
-#ifdef HYPERION_PANEL_3U
+  #ifdef HYPERION_PANEL_3U
```

Comment:
Should add a HYPERION_IS_STUBBED around this stuff

Reply:
Done